lib/AST/Identifier.cpp

```diff
@@ -20,13 +20,21 @@
 #include "llvm/Support/ConvertUTF.h"
 using namespace swift;
 
+void *DeclBaseName::SubscriptIdentifierData =
```

Comment:
Hm. Maybe I take it back about going from `void *` to `char *`. I was getting mixed up with levels of pointer, more so with the self-address assignment. And the `char *` better not be used as a string anyway (since there's no null byte).

The pointer needs to be 4-byte-aligned because you've claimed (and we rely on) DeclBaseName having two "spare bits" in its opaque representation. Using the address of a `void *` should handle that. :-)

The "conversions" between `void *` and `Identifier` would get optimized away in release builds, so I wouldn't have considered that such a blocker. PointerUnion does this all the time.